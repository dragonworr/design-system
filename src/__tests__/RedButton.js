import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { RedButton, theme } from '..'

describe('RedButton', () => {
  test('renders', () => {
    const json = renderer.create(<RedButton theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
  })
  
  test('disabled prop sets', () => {
    const json = renderer.create(<RedButton disabled theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.red)
  })

  test('without disabled prop sets', () => {
    const json = renderer.create(<RedButton theme={theme} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.darkRed, {
      modifier: ':hover',
    })
  })

})
