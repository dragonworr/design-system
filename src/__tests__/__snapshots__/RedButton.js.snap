// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RedButton disabled prop sets 1`] = `
.c1 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c1:disabled {
  opacity: 0.25;
}

.c0 {
  background-color: #c00;
}

<button
  className="c0 c1"
  disabled={true}
/>
`;

exports[`RedButton renders 1`] = `
.c1 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c1:disabled {
  opacity: 0.25;
}

.c1:hover {
  background-color: #049;
}

.c0 {
  background-color: #c00;
}

.c0:hover {
  background-color: #800;
}

<button
  className="c0 c1"
/>
`;

exports[`RedButton without disabled prop sets 1`] = `
.c1 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c1:disabled {
  opacity: 0.25;
}

.c1:hover {
  background-color: #049;
}

.c0 {
  background-color: #c00;
}

.c0:hover {
  background-color: #800;
}

<button
  className="c0 c1"
/>
`;
